django/db/models/sql/compiler.py

```diff
@@ -1934,11 +1921,11 @@ def execute_sql(self, returning_fields=None):
                         ),
                     )
                 ]
+            else:
+                return []
```

Comment:
Databases that don't have support for `RETURNING` and don't have an auto primary key can't return anything.